examples/demo-app/src/cloud-providers/carto-provider.js

```diff
@@ -0,0 +1,352 @@
+        result = await cs.updateVisualization({
+          id: this.currentMap.id,
+          name,
+          description,
+          thumbnail: thumbnailBase64,
+          config: JSON.stringify(config),
+          isPrivate: this.currentMap.isPrivate
+        }, cartoDatasets)
+      } else {
+        // TODO: Ask for changing current shared map generation because of being too oriented to file based clouds
+        // Check public name generation and replace
+        const regex = /(?:^keplergl_)([a-z0-9]+)(?:.json$)/;
+        const capturedName = name.match(regex);
+        const visName = capturedName ? `sharedmap_${capturedName[1]}` : name;
+
+        result = await cs.createVisualization({
+          name: visName,
+          description,
+          thumbnail: thumbnailBase64,
+          config: JSON.stringify(config),
+          isPrivate: !isPublic
+        }, cartoDatasets, true);
+      }
+
+      if (result) {
+        this.currentMap = result;
+      }
+
+      return ({
+        url: this._composeURL({
+          mapId: result.id,
+          owner: this._carto.username,
+          privateMap: !isPublic
+        })
+      });
+    } catch (error) {
+      this._manageErrors(error);
+    }
+  }
+
+  /**
+   * The CARTO cloud provider sets by the default the privacity to public
+   * @param {*} metadata
+   */
+  shareFile(metadata) {
+    return;
+  }
+
+  /**
+   * Returns the access token. If it has expired returns null. The toolkit library loads it
+   * from localStorage automatically
+   */
+  getAccessToken() {
+    let accessToken = null;
+    try {
+      accessToken = this._carto.oauth.expired ? null : this._carto.oauth.token;
+    } catch (error) {
+      this._manageErrors(error, false);
+    }
+
+    return accessToken;
+  }
+
+  getUserName(){
+    let username = null;
+    try {
+      username = this._carto.oauth.expired ? null : this._carto.username;
+    } catch (error) {
+      this._manageErrors(error, false);
+    }
+
+    return username;
+  }
+
+  /**
+   * The CARTO cloud provider polls the created window internally to parse the URL
+   * @param {*} location
+   */
+  getAccessTokenFromLocation(location) {
+    return;
+  }
+
+  async loadMap(queryParams) {
+    try {
+      const { owner: username, mapId, privateMap } = queryParams;
+
+      if (!username || !mapId) {
+        return;
+      }
+
+      let visualization;
+
+      if (privateMap.trim().toLowerCase() === 'true') {
+        await this._carto.login();
+        const currentUsername = this.getUserName();
+        if (currentUsername && currentUsername===username) {
+          const cs = await this._carto.getCustomStorage();
+          visualization = await cs.getVisualization(mapId);
+        }
+      } else {
+        visualization = await this._carto.PublicStorageReader.getVisualization(username, mapId);
+      }
+
+      if (!visualization) {
+        throw {target: {status: 404, responseText: `Can't find map with ID: ${mapId}`}};
```

Comment:
this error will be caught on line 231, and pass to `_maangeErrors`, in order to get the correct message, this should be throw ```new Error(`Can't find map with ID: ${mapId}`)```